lib/active_storage_validations/matchers/size_validator_matcher.rb

```diff
@@ -111,8 +111,9 @@ def not_higher_than_max?
 
       def passes_validation_with_size(size)
         mock_size_for(io, size) do
-          attach_file
-          validate
+          attach_file do
+            validate
+          end
```

Comment:
Could you do 
```ruby
attach_file
validate
remove_file
```
instead?

This fix logic can be replicated to other matchers in `has_many` case. Some can't use the `yield` logic like `AspectRatioValidatorMatcher`

Reply:
done

Reply:
Thanks a lot for the concern, it's definitely the way to go 👍